src/Sign.SignatureProviders.TrustedSigning/TrustedSigningService.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Azure;
+using Azure.CodeSigning;
+using Azure.CodeSigning.Models;
+using Azure.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Sign.Core;
+
+namespace Sign.SignatureProviders.TrustedSigning
+{
+    internal sealed class TrustedSigningService : ISignatureAlgorithmProvider, ICertificateProvider, IDisposable
+    {
+        private static readonly SignRequest _emptyRequest = new(SignatureAlgorithm.RS256, new byte[32]);
+
+        private readonly CertificateProfileClient _client;
+        private readonly string _accountName;
+        private readonly string _certificateProfileName;
+        private readonly ILogger<TrustedSigningService> _logger;
+        private readonly SemaphoreSlim _mutex = new(1);
+        private X509Certificate2? _publicKey;
+
+        public TrustedSigningService(
+            IServiceProvider serviceProvider,
+            TokenCredential tokenCredential,
+            Uri endpointUrl,
+            string accountName,
+            string certificateProfileName)
+        {
+            ArgumentNullException.ThrowIfNull(serviceProvider, nameof(serviceProvider));
+            ArgumentNullException.ThrowIfNull(tokenCredential, nameof(tokenCredential));
+            ArgumentNullException.ThrowIfNull(endpointUrl, nameof(endpointUrl));
+            ArgumentException.ThrowIfNullOrEmpty(accountName, nameof(accountName));
+            ArgumentException.ThrowIfNullOrEmpty(certificateProfileName, nameof(certificateProfileName));
+
+            _client = new CertificateProfileClient(tokenCredential, endpointUrl);
```

Comment:
Should this be a ctor parameter also, perhaps via an interface if available?

Reply:
There is no interface in this library otherwise I would have used that. I hope that a new library will give us an interface.

Reply:
Okay, gotcha. I guess it's a nitpick then. We have a separation of concerns with `TrustedSigningServiceProvider` so we can move the service initialization (`new`) and dependency injection (`GetRequiredService`) out of the ctor here.

I don't feel strongly, but it does seem to better align with my guessed purpose of `TrustedSigningServiceProvider`.

Reply:
As mentioned in my other comment I do think it's a good idea to remove the `GetRequiredService` call here. We could introduce a wrapper around the `CertificateProfileClient` class but I also wonder if we need to make that investment now when we will replace this with another TrustedSigning library once this becomes available.

Reply:
You could resolve `CertificateProfileClient` in the call site of this ctor and simply pass in the implementation as a parameter. I only call this out because I've generally been served well by following DI patterns as closely as possible even in CLI contexts. But I agree we can leave this as-is and re-evaluate as the dependent library changes. I'll leave it up to you on what to do.

Reply:
I hope that I can inject a client or a factory to create the client (because we need to specify `tokenCredential` and `endpointUrl`). That is why I have hidden this in the constructor for now. I will leave it up to the team to make the final decision.